### PR TITLE
Migrate search bundle documentation and associated example plugins

### PR DESCRIPTION
**Pull request by philipgatzka**
Affected article: shopware-5-search-bundle
Affected plugins: SwagSearchBundle, SwagAttributeFilter

### CODE REVIEW

File: source/developers-guide/shopware-5-search-bundle/index.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ This small example shows how the product number search can be used.
 
 ## Reference to SearchBundleDBAL
 As already described, the `Shopware\Bundle\SearchBundle` defines only how product lists are selected and which conditions and sorting criteria are used to restrict or modify the result set.
-The `SearchBundle` contains no specify search engine implementation, like `Doctrine\ORM` or `PDO`, and can't be used as standalone search mechanism.
+The `SearchBundle` contains no search engine implementation, like `Doctrine\ORM` or `PDO`, and can't be used as standalone search mechanism.
```
Comment (mitelg) [inline source/developers-guide/shopware-5-search-bundle/index.md]: ... no specific search ...

File: exampleplugins/5.2/SwagAttributeFilter/Subscriber/AttributeFilterSubscriber.php [hunk 2]
```diff
@@ -36,11 +36,20 @@ public static function getSubscribedEvents()
         return [
             'Shopware_SearchBundleDBAL_Collect_Facet_Handlers' => 'registerFacetHandler',
```
Comment (mitelg) [inline exampleplugins/5.2/SwagAttributeFilter/Subscriber/AttributeFilterSubscriber.php]: please use the service tags in the `services.xml` to register new facetHandler, conditionHandler and CriteriaRequestHandler. 
you should rewrite the whole plugin and the article

Comment (philipgatzka) [reply exampleplugins/5.2/SwagAttributeFilter/Subscriber/AttributeFilterSubscriber.php]: Done

File: source/developers-guide/shopware-5-search-bundle/index.md (hunk 1 above)
Comment (philipgatzka) [reply source/developers-guide/shopware-5-search-bundle/index.md]: Done

File: source/developers-guide/shopware-5-search-bundle/index.md [hunk 3]
```diff
@@ -140,20 +140,44 @@ Each `ConditionHandler` must implement the `Shopware\Bundle\SearchBundleDBAL\Con
...
-);
+
+namespace YourPlugin;
+
+class YourPlugin extends \Shopware\Components\Plugin
+{
+    public static function getSubscribedEvents()
+    {
+        return [
+            'Shopware_SearchBundleDBAL_Collect_Condition_Handlers' => 'onCollectConditionHandlers'
```
Comment (mitelg) [inline source/developers-guide/shopware-5-search-bundle/index.md]: please do not describe the collect event, but the `tag` in the services.xml instead

**Comment by mitelg** [comment]
please rework all plugins. 
- do not use the collect events for registering ConditionHandlers, FacetHandlers, etc
- use a service declaration and the specific tag to register the handlers
- consider this in all plugins and in the text

**Pull request merged**